models/articles.model.js

```diff
@@ -2,16 +2,31 @@ const { table } = require("console");
 const fs = require("fs/promises");
 const db = require("../db/connection");
 
-exports.findArticles = () => {
+exports.findArticles = (topic) => {
+    if(topic){
     const sql = 
     `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url,
     COUNT(comments.article_id) AS comment_count
     FROM articles
-    JOIN comments on articles.article_id = comments.article_id
+    LEFT JOIN comments on articles.article_id = comments.article_id
+    WHERE topic = $1
+    GROUP BY articles.article_id
+    ORDER BY created_at DESC`
+    return db.query(sql, [topic])
+    .then((result) => {
+        return result.rows
+    });
+} else { 
+    const sql = 
+    `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url,
+    COUNT(comments.article_id) AS comment_count
+    FROM articles
+    LEFT JOIN comments on articles.article_id = comments.article_id
     GROUP BY articles.article_id
```

Comment:
This will work but it's not DRY code, have a look at this if you want to refactor in the future: https://notes.northcoders.com/courses/js-back-end/complex-queries#optional-where-clauses